### ISSUE TITLE
client_intf.so: undefined symbol: sc_get_auth_oid

### ISSUE THREAD

**Issue by ArthurName** [net-snmp/net-snmp#363]
**net-snmp**
5.9.1

**System**
DISTRIB_ID=Ubuntu
DISTRIB_RELEASE=18.04
DISTRIB_CODENAME=bionic
DISTRIB_DESCRIPTION="Ubuntu 18.04.6 LTS"
(not from scratch; upgraded from 16.04)

**Goal**
My goal is to get netsnmp-python installed in my virtual Python environment (pyenv + pyenv-virtualenv), i.e. I want pip freeze to report that **netsnmp-python==1.0a1** is installed, within the virtual environment. So, if this issue report indicates I'm doing something unnecessary or incorrect, I'm all ears. _N.b. it's an environment for Python 2.7, which is a requirement. And I've used net-snmp v5.9.1 here, but if needed I could do with v5.7.3 or whatever is rational._

**What I did**
`./configure --with-defaults`
[- - -]
```
---------------------------------------------------------
            Net-SNMP configuration summary:
---------------------------------------------------------

  SNMP Versions Supported:    1 2c 3
  Building for:               linux
  Net-SNMP Version:           5.9.1
  Network transport support:  Callback Unix Alias TCP UDP TCPIPv6 UDPIPv6 IPv4Base SocketBase TCPBase UDPIPv4Base UDPBase IPBase IPv6Base
  SNMPv3 Security Modules:     usm
  Agent MIB code:            default_modules =>  snmpv3mibs mibII ucd_snmp notification notification-log-mib target agent_mibs agentx disman/event disman/schedule utilities host
  MYSQL Trap Logging:         unavailable
  Embedded Perl support:      enabled
  SNMP Perl modules:          building -- embeddable
  SNMP Python modules:        disabled
  Crypto support from:        crypto
  Authentication support:     MD5 SHA1 SHA224 SHA256 SHA384 SHA512
  Encryption support:         DES AES AES128 AES192 AES192C AES256 AES256C
  Local DNSSEC validation:    disabled

---------------------------------------------------------
```

`make`
(according to INSTALL file, but fails:)
[- - -]
```
Assembler messages:
Fatal error: can't create ucd-snmp/.libs/disk_hw.o: Permission denied
Makefile:100: recipe for target 'ucd-snmp/disk_hw.lo' failed
make[2]: *** [ucd-snmp/disk_hw.lo] Error 1
make[2]: Leaving directory '/home/ANONYMOUS/FOO/BAR/X/Y/net-snmp-5.9.1/agent/mibgroup'
Makefile:1547: recipe for target 'subdirs' failed
make[1]: *** [subdirs] Error 1
make[1]: Leaving directory '/home/ANONYMOUS/FOO/BAR/X/Y/net-snmp-5.9.1/agent'
Makefile:674: recipe for target 'subdirs' failed
make: *** [subdirs] Error 1
```
So, instead:
`sudo make`
... which went okay.

Then
```
sudo make install
cd python
python setup.py build
```

And
`python setup.py test`
... with output:
```
running test
WARNING: Testing via this command is deprecated and will be removed in a future version. Users looking for a generic test entry point $
running egg_info
writing netsnmp_python.egg-info/PKG-INFO
writing top-level names to netsnmp_python.egg-info/top_level.txt
writing dependency_links to netsnmp_python.egg-info/dependency_links.txt
reading manifest file 'netsnmp_python.egg-info/SOURCES.txt'
writing manifest file 'netsnmp_python.egg-info/SOURCES.txt'
running build_ext
copying build/lib.linux-x86_64-2.7/netsnmp/client_intf.so -> netsnmp
Traceback (most recent call last):
  File "setup.py", line 43, in <module>
    libraries=libs )
  File "/home/ANON/.pyenv/versions/FOO2713/lib/python2.7/site-packages/setuptools/__init__.py", line 162, in setup
    return distutils.core.setup(**attrs)
  File "/home/ANON/.pyenv/versions/2.7.13/lib/python2.7/distutils/core.py", line 151, in setup
    dist.run_commands()
  File "/home/ANON/.pyenv/versions/2.7.13/lib/python2.7/distutils/dist.py", line 953, in run_commands
    self.run_command(cmd)
  File "/home/ANON/.pyenv/versions/2.7.13/lib/python2.7/distutils/dist.py", line 972, in run_command
    cmd_obj.run()
  File "/home/ANON/.pyenv/versions/FOO2713/lib/python2.7/site-packages/setuptools/command/test.py", line 237, in run
    self.run_tests()
  File "/home/ANON/.pyenv/versions/FOO2713/lib/python2.7/site-packages/setuptools/command/test.py", line 259, in run_tests
    exit=False,
  File "/home/ANON/.pyenv/versions/2.7.13/lib/python2.7/unittest/main.py", line 94, in __init__
    self.parseArgs(argv)
  File "/home/ANON/.pyenv/versions/2.7.13/lib/python2.7/unittest/main.py", line 149, in parseArgs
    self.createTests()
  File "/home/ANON/.pyenv/versions/2.7.13/lib/python2.7/unittest/main.py", line 158, in createTests
    self.module)
  File "/home/ANON/.pyenv/versions/2.7.13/lib/python2.7/unittest/loader.py", line 130, in loadTestsFromNames
    suites = [self.loadTestsFromName(name, module) for name in names]
  File "/home/ANON/.pyenv/versions/2.7.13/lib/python2.7/unittest/loader.py", line 91, in loadTestsFromName
    module = __import__('.'.join(parts_copy))
  File "/home/ANON/git/FOO/src/lib/net-snmp-5.9.1/python/netsnmp/__init__.py", line 1, in <module>
    from .client import *
  File "/home/ANON/git/FOO/src/lib/net-snmp-5.9.1/python/netsnmp/client.py", line 5, in <module>
    import netsnmp.client_intf
ImportError: /home/ANON/git/FOO/src/lib/net-snmp-5.9.1/python/netsnmp/client_intf.so: undefined symbol: sc_get_auth_oid
```

I install anyway and it works:
`python setup.py install`

```
pip freeze | grep snmp
netsnmp-python==1.0a1
```
But as expected I guess, it does not work:
```
Python 2.7.13 (default, Oct 12 2021, 14:59:08) 
[GCC 7.5.0] on linux2
Type "help", "copyright", "credits" or "license" for more information.
>>> import netsnmp
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "build/bdist.linux-x86_64/egg/netsnmp/__init__.py", line 1, in <module>
  File "build/bdist.linux-x86_64/egg/netsnmp/client.py", line 5, in <module>
  File "build/bdist.linux-x86_64/egg/netsnmp/client_intf.py", line 7, in <module>
  File "build/bdist.linux-x86_64/egg/netsnmp/client_intf.py", line 6, in __bootstrap__
ImportError: /home/ANON/.cache/Python-Eggs/netsnmp_python-1.0a1-py2.7-linux-x86_64.egg-tmp/netsnmp/client_intf.so: undefined symbol: sc_get_auth_oid
```

**To sum up**
1. Why the permission denied? Where does it try to put that file? (Is this relevant?)
2. Why the undefined symbol?
